soundbay/inference.py

```diff
@@ -109,36 +112,60 @@ def inference_to_file(
         model_path: directory for the wanted trained model
         output_path: directory to save the prediction file
     """
-    # set paths and create dataset
-    test_dataset = instantiate(dataset_args)
-
-    # load model
-    model = load_model(model_args, checkpoint_state_dict).to(device)
-
-    test_dataloader = DataLoader(dataset=test_dataset, shuffle=False, batch_size=batch_size, num_workers=0,
-                                 pin_memory=False)
-
-    # predict
-    predict_prob = predict_proba(model, test_dataloader, device, None)
-
-    results_df = pandas.DataFrame(predict_prob)
-    if hasattr(test_dataset, 'metadata'):
-        concat_dataset = pandas.concat([test_dataset.metadata, results_df], axis=1) #TODO: make sure metadata column order matches the prediction df order
-        metrics_dict = Logger.get_metrics_dict(concat_dataset["label"].values.tolist(),
-                                               np.argmax(predict_prob, axis=1).tolist(),
-                                               results_df.values)
-        print(metrics_dict)
-    else:
-        concat_dataset = results_df
-        print("Notice: The dataset has no ground truth labels")
+    # Check how many channels
+    data_sample, _ = sf.read(dataset_args.file_path, start=0, stop=10)
+
+    all_channel_list = []
+    all_channel_raven_list = []
+    for channel in range(data_sample.shape[1]):
+        # set paths and create dataset
+        test_dataset = instantiate(dataset_args, channel=channel)
+
+        # load model
+        model = load_model(model_args, checkpoint_state_dict).to(device)
+
+        test_dataloader = DataLoader(dataset=test_dataset, shuffle=False, batch_size=batch_size, num_workers=0,
+                                     pin_memory=False)
+
+        # predict
+        predict_prob = predict_proba(model, test_dataloader, device, None)
+
+        results_df = pandas.DataFrame(predict_prob)
+        if hasattr(test_dataset, 'metadata'):
+            concat_dataset = pandas.concat([test_dataset.metadata, results_df], axis=1) #TODO: make sure metadata column order matches the prediction df order
+            metrics_dict = Logger.get_metrics_dict(concat_dataset["label"].values.tolist(),
+                                                   np.argmax(predict_prob, axis=1).tolist(),
+                                                   results_df.values)
+            print(metrics_dict)
+        else:
+            concat_dataset = results_df
+            print("Notice: The dataset has no ground truth labels")
+
+        # create raven file
+        if save_raven:
+            all_channel_raven_list.append(
+                inference_csv_to_raven(results_df, predict_prob.shape[1], dataset_args.seq_length, 1, threshold, 'call',
```

Comment:
fixed

Reply:
Hey there.
I think we still have a problem - really small fix is needed; single tab is missing to make the hierarchy right in `inference_file.yaml` OR using `data.seq_length` instead of `data.test_dataset.seq_length` in the code snippet that is linked to the comment.
I know this PR has already been merged but I am writing here for documentation purposes